application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPagesMacro/RowEvolutionMostViwedPagesJSON.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.1" encoding="UTF-8"?>
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.4" reference="Analytics.Code.Macros.MostViewedPagesMacro.RowEvolutionMostViewedPagesJSON" locale="">
+  <web>Analytics.Code.Macros.MostViewedPagesMacro</web>
+  <name>RowEvolutionMostViewedPagesJSON</name>
```

Comment:
It a little hard because you don't know from the start what you will need for other macros. But, for example, this `RowEvolutionMostViewedPagesJSON` doesn't need to be that specific. You could have a `RowEvolutionJSON` and use some request parameter to modify what JSON response it gives